test/unit/krane/kubectl_test.rb

```diff
@@ -348,6 +348,29 @@ def test_retry_delay_backoff
     end
   end
 
+  def test_kubectl_run_fixes_encoding_when_locales_set_to_non_utf8
```

Comment:
This is now a much better version of the test that reproduces the Ruby environment from the bug report (to confirm, start a console session with the wrong local envs set and check the `Encoding` values below). Implementation inspired by [this patch on Pysch](https://github.com/ruby/psych/commit/fe65329ce9ece399d61dadf1610e342ff456654e), which I found via [a bug report on RubyGems](https://github.com/rubygems/rubygems/issues/139).

Note that I confirmed this is a true regression test that files if I remove my new code.